src/tdastro/astro_utils/passbands.py

```diff
@@ -2,257 +2,246 @@
+
+    def fluxes_to_bandfluxes(self, flux_density_matrix, wavelengths_angstrom) -> np.ndarray:
+        """Calculate bandfluxes for all passbands in the group.
 
         Parameters
         ----------
-        band_name : str
-            Name of the band for which the transmission table is to be downloaded.
-        output_file : str
-            Path to the file where the transmission table will be saved.
-        timeout : int
-            Time in seconds to wait for the file to be downloaded and created.
+        flux_density_matrix : np.ndarray
+            A 2D array of flux densities where rows are times and columns are wavelengths.
+        wavelengths_angstrom : np.ndarray
+            An array of wavelengths in Angstroms corresponding to the columns of flux_density_matrix.
 
         Returns
         -------
-        bool
-            True if the transmission table is successfully downloaded, False otherwise.
+        dict
+            A dictionary of bandfluxes with passband labels as keys.
         """
-        try:
-            socket.setdefaulttimeout(timeout)
+        bandfluxes = {}
+        for label, passband in self.passbands.items():
+            bandfluxes[label] = passband.fluxes_to_bandflux(flux_density_matrix, wavelengths_angstrom)
+        return bandfluxes
 
-            url = (
-                f"http://svo2.cab.inta-csic.es/svo/theory/fps3/getdata.php"
-                f"?format=ascii&id=LSST/LSST.{band_name}"
-            )
 
-            urllib.request.urlretrieve(url, output_file)
+class Passband:
+    """A passband contains information about its transmission curve and calculates its normalization.
 
-            if os.path.getsize(output_file) == 0:
-                logging.error(f"Transmission table for {band_name} is empty.")
-                return False
+    Attributes
+    ----------
+    label : str
+        The label of the passband.
+     survey : str
+        The survey to which the passband belongs.
+    full_name : str
+        The full name of the passband.
+    table_path : str
+        The path to the transmission table file.
+    table_url : str
+        The URL to download the transmission table file.
+    transmission_table : np.ndarray
+        A 2D array where the first col is wavelengths (Angstrom) and the second col is transmission values.
+    normalized_transmission_table : np.ndarray
+        A 2D array of wavelengths and normalized transmissions.
+    """
 
-            logging.info(f"Downloaded: {band_name}-band transmission table.")
-            return True
-        except HTTPError as e:
-            logging.error(f"HTTP error occurred when downloading transmission table for {band_name}: {e}")
-            return False
-        except URLError as e:
-            logging.error(f"URL error occurred when downloading transmission table for {band_name}: {e}")
-            return False
+    def __init__(self, survey, label, table_path=None, table_url=""):
```

Comment:
Would it be useful to have the option of passing in the transmission table as a numpy array manually? I can imagine cases where that would be useful for testing, but it might not be needed in normal runs of the code.

Reply:
I was on the fence about that--ultimately I chose to make the unit tests rather than the code be slightly more complicated, but would be happy to switch if that's preferred.